service/controller/chart/resource/status/create.go

```diff
@@ -89,7 +89,10 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		Version: releaseContent.Version,
 	}
 	if !releaseContent.LastDeployed.IsZero() {
-		desiredStatus.Release.LastDeployed = &metav1.Time{Time: releaseContent.LastDeployed}
+		// We convert the timestamp to the nearest second to match the value in
+		// the chart CR status.
+		lastDeployed := releaseContent.LastDeployed.Unix()
+		desiredStatus.Release.LastDeployed = &metav1.Time{Time: time.Unix(lastDeployed, 0)}
```

Comment:
The last deployed timestamp we get from helm is to the nearest nanosecond. But the k8s API returns it to the nearest second.

This is just for info so the nearest second is fine.

Reply:
What do you think about adding a new unittest for comparing nanoseconds?

Reply:
Yes good idea. I updated `equals` to compare using the unix epoch which is to the nearest second. It just needed some handling in case its nil.

Reply:
Looks good to me :)